opengever/maintenance/scripts/repository_migration.py

```diff
@@ -765,18 +764,34 @@ def validate_operation(self, operation):
                     "\nInvalid operation: blocking inheritance without setting "
                     "local roles. {}\n".format(operation))
                 operation['is_valid'] = False
-            if has_local_roles:
-                obj = unrestrictedUuidToObject(operation['uid'])
-                if obj:
-                    # newly created positions will have the local_roles set
-                    # in the pipeline
-                    operation['set_permissions'] = True
-                if obj and RoleAssignmentManager(obj).get_assignments_by_cause(ASSIGNMENT_VIA_SHARING):
+            obj = unrestrictedUuidToObject(operation['uid'])
+            if not obj:
+                # newly created positions will have the local_roles set
+                # in the pipeline
+                pass
+            elif self.are_permissions_updated(obj, permissions, inheritance_blocked):
+                operation['set_permissions'] = True
+                if RoleAssignmentManager(obj).get_assignments_by_cause(ASSIGNMENT_VIA_SHARING):
                     operation['local_roles_deleted'] = True
                     logger.warning(
                         "\nSharing assignments for {} will be deleted and "
                         "replaced.\n".format(obj.absolute_url_path()))
 
+    def are_permissions_updated(self, obj, permissions, inheritance_blocked):
```

Comment:
I thought it means if that if the permissions are already updated... what do you think about `needs_permission_update`?